stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -180,7 +180,12 @@ function(_add_target_variant_c_compile_flags)
 
   is_build_type_optimized("${CFLAGS_BUILD_TYPE}" optimized)
   if(optimized)
-    list(APPEND result "-O2")
+    is_build_type_optimized_for_size("${CFLAGS_BUILD_TYPE}" optimized_for_size)
+    if(optimized_for_size)
+      list(APPEND result "-Os")
+    else()
```

Comment:
This code repeating is sort of unfortunate. Maybe refactor it into a helper?

Reply:
Something like `clang_optimize_flag_for_build_type()`?

Reply:
I would be fine with that.